archive/p/python/game-of-life.py

```diff
@@ -1,108 +1,120 @@
-                cell.neighbors.append(self.grid[previous_row][next_col])
-                cell.neighbors.append(self.grid[next_row][previous_col])
-                cell.grid(column=curr_col, row=curr_row)
-
-    def generate(self):
-        self._populate()
-        self._link()
-        self.after(self.delay, self.step)
-
-    def step(self):
-        for row in self.grid:
-            for cell in row:
-                cell.transition()
-                cell['bg'] = cell.get_background_color()
-        for row in self.grid:
-            for cell in row:
-                cell.clear_state()
-        self.frame_index += 1
-        if self.frame_index < self.total_frames:
-            self.after(self.delay, self.step)
-
-
-def main():
-    if len(sys.argv) > 4:
-        width = int(sys.argv[1])
-        frame_rate = float(sys.argv[2])
-        total_frames = int(sys.argv[3])
-        spawn_rate = float(sys.argv[4])
-        my_grid = Grid(width, spawn_rate, frame_rate, total_frames)
-    else:
-        my_grid = Grid()
-    my_grid.generate()
-    my_grid.mainloop()
-
-
-if __name__ == '__main__':
-    main()
+root = Tk()
+root.title("The Renegade Coder's Game of Life")
+frame = Frame(root, width=720, height=720)
+frame.pack()
+canvas = Canvas(frame, width=720, height=720)
+canvas.pack()
+
+def create_grid():
+	x = 10
+	y = 10
+	global grid # Variable to store the Cell objects
+	global rectangles # Variable to store rectangles
+	rectangles = []
+	grid = []
+	for i in range(70):
+		grid.append([])
+		rectangles.append([])
+		for j in range(70):
+			rect = canvas.create_rectangle(x, y, x+10, y+10, fill="white")
+			rectangles[i].append(rect)
+			grid[i].append(Cell(x, y, i, j))
+			x += 10
+		x = 10
+		y += 10
+
+def rectangle_coordinates(x, y):
+	return (x- x%10, y - y%10)
+
+def click_colour_change(event):
+	print (event.x, event.y)
+	x, y = rectangle_coordinates(event.x, event.y)
+	try:
+		iy = x / 10 - 1
+		ix = y / 10 - 1
+		if ix == -1 or iy == -1:
+			raise IndexError
+			if grid[ix][iy].isAlive:
+				canvas.itemconfig(rectangles[ix][iy], fill="black")
+			else:
+				canvas.itemconfig(rectangles[ix][iy], fill="white")
+				grid[ix][iy].switchStatus()
+				print (grid[ix][iy].pos_matrix, grid[ix][iy].pos_screen)
+	except IndexError:
+		return
+
+def paint_grid():
+	for i in grid:
+		for j in i:
+			if j.nextStatus != j.isAlive:
+				x, y = j.pos_matrix
+				print (x, y)
+				if j.nextStatus:
+					canvas.itemconfig(rectangles[x][y], fill="white")
+					print ("changed", j.pos_matrix, "from dead to alive")
+				else:
+					canvas.itemconfig(rectangles[x][y], fill="black")
+					print ("changed", j.pos_matrix, "from alive to dead")
+					j.switchStatus()
+					print ("Current status of", j.pos_matrix, j.isAlive)
+
+def status_update(cell):
+	num_alive = 0
+	x, y = cell.pos_matrix
+	for i in (x-1, x, x+1):
+		for j in (y-1, y, y+1):
+			if i == x and j == y:
+				continue
+				if i == -1 or j == -1:
+					continue
+					try:
```

Comment:
Here's another section of unreachable code. I believe the continue will jump to the next iteration of the loop, but don't quote me on that. 😆